utils/newsletter.js

```diff
@@ -139,11 +141,14 @@ export default class Newsletter {
 
         try {
             const template = await ejs.renderFile(templatePath, renderingData);
-            return injectUrlTracking
+            const injectedHtml = injectUrlTracking
                 ? await this.#injectUrlTracking(renderingData, template)
-                : {trackedLinks: [], modifiedHtml: template};
+                : {trackedLinks: new Set(), modifiedHtml: template};
+
+            // add widgets, inline css and minify the html.
+            return await Widgets.replace(renderingData, injectedHtml);
         } catch (error) {
-            logError(logTags.Newsletter, error);
+            logError(logTags.Newsletter, error.message);
```

Comment:
This needs to be `error` only. Seems like I missed changing it back post some testing.